app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/notes/NotesPresenter.kt

```diff
@@ -0,0 +1,89 @@
+package org.dhis2.usescases.teiDashboard.dashboardfragments.notes
+
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.processors.FlowableProcessor
+import io.reactivex.processors.PublishProcessor
+import org.dhis2.data.schedulers.SchedulerProvider
+import org.dhis2.data.tuples.Pair
+import org.dhis2.usescases.teiDashboard.DashboardRepository
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.enrollment.EnrollmentStatus
+import org.hisp.dhis.android.core.maintenance.D2Error
+import org.hisp.dhis.android.core.note.NoteCreateProjection
+import timber.log.Timber
+
+class NotesPresenter(
+    val d2: D2,
+    val dashboardRepository: DashboardRepository,
+    val schedulerProvider: SchedulerProvider,
+    val view: NotesContracts.View
+) {
+
+    var compositeDisposable: CompositeDisposable = CompositeDisposable()
+    var processor: FlowableProcessor<Boolean> = PublishProcessor.create()
+    private lateinit var programUid: String
+    private lateinit var teiUid: String
+
+    fun init(programUid: String, teiUid: String) {
+        this.programUid = programUid
+        this.teiUid = teiUid
+    }
```

Comment:
Maybe we can pass theses parameters in the `subscribeToNotes()` method which is called in the `onResume()` method of the fragment.

I'm not sure a `init()` is necessary in this scenario.